src/Functions/conv.cpp

```diff
@@ -0,0 +1,238 @@
+            number_column = castColumn(string_arg, std::make_shared<DataTypeString>());
+        }
+        const auto * number_col = checkAndGetColumn<ColumnString>(number_column.get());
+        const auto * number_const_col = checkAndGetColumnConst<ColumnString>(number_column.get());
+        if (!number_col && !number_const_col)
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Illegal column {} of first argument of function {}",
+                arguments[0].column->getName(),
+                getName());
+        const auto & from_base_column = arguments[1].column;
+        const auto & to_base_column = arguments[2].column;
+        auto result_column = ColumnString::create();
+        auto & result_data = result_column->getChars();
+        auto & result_offsets = result_column->getOffsets();
+        result_offsets.resize(input_rows_count);
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            std::string number_str;
+            if (number_col)
+                number_str = number_col->getDataAt(i).toString();
+            else
+                number_str = number_const_col->getValue<String>();
+            Int64 from_base = from_base_column->getInt(i);
+            Int64 to_base = to_base_column->getInt(i);
+            std::string result = convertNumber(number_str, static_cast<int>(from_base), static_cast<int>(to_base));
+            result_data.insert(result.begin(), result.end());
+            result_data.push_back(0);
+            result_offsets[i] = result_data.size();
+        }
+
+        return result_column;
+    }
+
+private:
+    static std::string convertNumber(const std::string & number, int from_base, int to_base)
+    {
+        if (from_base < 2 || from_base > 36 || to_base < 2 || to_base > 36)
+            return "";
+        if (number.empty())
+            return "";
+        std::string clean_number = number;
+        bool is_negative = false;
+        // Trim leading whitespace
+        clean_number.erase(0, clean_number.find_first_not_of(" \t\n\r\f\v"));
+        // Trim trailing whitespace
+        if (!clean_number.empty())
+        {
+            clean_number.erase(clean_number.find_last_not_of(" \t\n\r\f\v") + 1);
+        }
+        if (clean_number.empty())
+            return "";
+
+        if (clean_number[0] == '-')
+        {
+            is_negative = true;
+            clean_number = clean_number.substr(1);
+        }
+        else if (clean_number[0] == '+')
+        {
+            clean_number = clean_number.substr(1);
+        }
+
+        if (clean_number.empty())
+            return "";
+
+        try
+        {
+            UInt64 value = parseFromBase(clean_number, from_base);
+            if (is_negative)
+            {
+                value = static_cast<UInt64>(-static_cast<Int64>(value));
+            }
+            return formatToBase(value, to_base);
+        }
+        catch (...)
```

Comment:
I think we should try log exception, or better throw an exception for such cases.

Reply:
sure, I will simply throw an exception, what kind of exception shall I throw?

Reply:
Perhaps just `ILLEGAL_TYPE_OF_ARGUMENT` or another suitable code from [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/ErrorCodes.cpp).